Source/SwiftLintFramework/Rules/LegacyConstantRule.swift

```diff
@@ -15,66 +15,63 @@ public struct LegacyConstantRule: CorrectableRule, ConfigurationProviderRule {
-            "↓CGRectNull"
-        ],
-        corrections: [
-            "↓CGRectInfinite\n": "CGRect.infinite\n",
-            "↓CGPointZero\n": "CGPoint.zero\n",
-            "↓CGRectZero\n": "CGRect.zero\n",
-            "↓CGSizeZero\n": "CGSize.zero\n",
-            "↓NSZeroPoint\n": "NSPoint.zero\n",
-            "↓NSZeroRect\n": "NSRect.zero\n",
-            "↓NSZeroSize\n": "NSSize.zero\n",
-            "↓CGRectInfinite\n↓CGRectNull\n": "CGRect.infinite\nCGRect.null\n"
-        ]
-    )
+    public static let description: RuleDescription = {
+        let nonTriggeringExamples: [String]
+        let triggeringExampes: [String]
+        let corrections: [String: String]
+        switch SwiftVersion.current {
+        case .two:
+            nonTriggeringExamples = LegacyConstantRuleExamples.swift2NonTriggeringExamples
+            triggeringExampes = LegacyConstantRuleExamples.swift2TriggeringExamples
+            corrections = LegacyConstantRuleExamples.swift2Corrections
+        case .three:
+            nonTriggeringExamples = LegacyConstantRuleExamples.swift3NonTriggeringExamples
+            triggeringExampes = LegacyConstantRuleExamples.swift3TriggeringExamples
+            corrections = LegacyConstantRuleExamples.swift3Corrections
+        }
 
-    public func validate(file: File) -> [StyleViolation] {
-        let constants = ["CGRectInfinite", "CGPointZero", "CGRectZero", "CGSizeZero",
-                         "NSZeroPoint", "NSZeroRect", "NSZeroSize", "CGRectNull"]
+        return RuleDescription(
+            identifier: "legacy_constant",
+            name: "Legacy Constant",
+            description: "Struct-scoped constants are preferred over legacy global constants.",
+            nonTriggeringExamples: nonTriggeringExamples,
+            triggeringExamples: triggeringExampes,
+            corrections: corrections
+        )
+    }()
 
-        let pattern = "\\b(" + constants.joined(separator: "|") + ")\\b"
+    private static let legacyConstants: [String] = {
+        switch SwiftVersion.current {
+        case .two:
+            return Array(LegacyConstantRuleExamples.swift2Patterns.keys)
+        case .three:
+            return Array(LegacyConstantRuleExamples.swift3Patterns.keys)
+        }
+    }()
 
-        return file.match(pattern: pattern, with: [.identifier]).map {
-            StyleViolation(ruleDescription: type(of: self).description,
-                severity: configuration.severity,
-                location: Location(file: file, characterOffset: $0.location))
+    private static let legacyPatterns: [String: String] = {
+        switch SwiftVersion.current {
+        case .two:
+            return LegacyConstantRuleExamples.swift2Patterns
+        case .three:
+            return LegacyConstantRuleExamples.swift3Patterns
         }
+    }()
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "\\b(" + LegacyConstantRule.legacyConstants.joined(separator: "|") + ")"
```

Comment:
I know this was already in this way, but you can use a non-capture group here for a performance boost